lib/approval.rb

```diff
@@ -18,23 +18,23 @@ def self.call
     yield approval_api
   rescue ApprovalApiClient::ApiError => e
     Rails.logger.error("ApprovalApiClient::ApiError #{e.message}")
-    raise Catalog::ApprovalError, e.message
+    raise Catalog::ApprovalError.new(e.code, e.message)
   end
 
   def self.call_action_api
     pass_thru_headers
     yield action_api
   rescue ApprovalApiClient::ApiError => e
     Rails.logger.error("ApprovalApiClient::ApiError #{e.message}")
-    raise Catalog::ApprovalError, e.message
+    raise Catalog::ApprovalError.new(e.code, e.message)
```

Comment:
@lindgrenj6 This should also be an add, since we are trying to bubble up the exception should the e.add raise the exception. Maybe e.add!(code, message, e)